content/en/blog/2022-11-16-templating-and-diffs-with-kustomize.md

```diff
@@ -0,0 +1,337 @@
+  default/Service/podtato-head-entry
+  default/Service/podtato-head-hat
+  default/Service/podtato-head-left-arm
+  default/Service/podtato-head-left-leg
+  default/Service/podtato-head-right-arm
+  default/Service/podtato-head-right-leg
+  default/ServiceAccount/podtato-head
+```
+
+It will first show you what it will do, then ask for confirmation and finally show what it did. Verify that it got
+deployed:
+
+```shell
+$ kubect get pod
+NAME                                      READY   STATUS    RESTARTS   AGE
+podtato-head-entry-7dfd8cdd6d-6mtxd       1/1     Running   0          95s
+podtato-head-hat-6bcbf5f957-mfc6r         1/1     Running   0          95s
+podtato-head-left-arm-7d9db78544-689tx    1/1     Running   0          95s
+podtato-head-left-leg-59f45ffc4-grcjc     1/1     Running   0          95s
+podtato-head-right-arm-5444b48b85-427w7   1/1     Running   0          95s
+podtato-head-right-leg-f68df999f-g27nz    1/1     Running   0          95s
+```
+
+There are multiple things that you might have noticed already:
+1. Kustomize does not handle deployments, it just "builds" them and then lets you handle the actual deployment via `kubectl apply -f`.
+Kluctl also handles the actual deployment for you. The advantages of this will be clear in a few minutes.
+2. Kluctl showed a diff (very simple one in this case, just new objects), then asked for confirmation and then showed what it did (identical to what the diff showed).
+The power of the diff feature will become much clearer in a few minutes.
+
+## Introducing some templating
+
+Now let's introduce some templating into the example deployment. For example, let's edit `deployment-entry.yaml` and
+change the replicas field to:
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: podtato-head-entry
+...
+spec:
+  replicas: {{ args.entry_replicas }}
+...
+```
+
+If you'd try to re-deploy this now, it will throw an error about `args.entry_replicas` being undefined. So let's call
+Kluctl with the now required arg:
+
+```shell
+$ kluctl deploy -a entry_replicas=3
+...
+
+Changed objects:
+  default/Deployment/podtato-head-entry
+
+Diff for object default/Deployment/podtato-head-entry
++---------------+--------------------------------------------------------------+
+| Path          | Diff                                                         |
++---------------+--------------------------------------------------------------+
+| spec.replicas | -1                                                           |
+|               | +3                                                           |
++---------------+--------------------------------------------------------------+
+? The diff succeeded, do you want to proceed? (y/N)
+```
+
+You'll notice that Kluctl again stops and asks for confirmation. But this time, it will actually show you some
+meaningful diff. It allows you to verify that what Kluctl is going to do is actually what you intended to do. The diff
```

Comment:
```suggestion
meaningful diff. It allows you to verify that Kuctl will apply the intended changes. The diff
```

Reply:
Can you change the suggestion to with Kuctl -> Kluctl? :)

Reply:
Ah cool, just realized that I can edit your suggestion by myself :)